File: tutocpp_pkg/src/server_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

class ServerNode : public rclcpp::Node
{
public:
    ServerNode() : Node("server_node")
    {
        server_ = create_service<example_interfaces::srv::AddTwoInts>("add_ints", std::bind(&ServerNode::client_callback, this, std::placeholders::_1, std::placeholders::_2));
        RCLCPP_INFO(get_logger(), "Server Node has been started");
    }

private:
    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;
    void client_callback(example_interfaces::srv::AddTwoInts::Request::SharedPtr request, example_interfaces::srv::AddTwoInts::Response::SharedPtr response)
    {
        response->sum = request->a + request->b;
        RCLCPP_INFO(get_logger(), "%ld + %ld = %ld", request->a, request->b, response->sum);
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<ServerNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: tutopy_pkg/tutopy_pkg/server_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.srv import AddTwoInts


class ServerNode(Node):
    def __init__(self):
        super().__init__("server_node")
        self.create_service(AddTwoInts, "add_ints", self.add_callback)
        self.get_logger().info("Server Node has been started")

    def add_callback(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info(f"{request.a} + {request.b} = {response.sum}")
        return response


def main(args=None):
    rclpy.init(args=args)

    node = ServerNode()

    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
